File: backend/CustomerSupport.Domain/Entities/Ticket.cs
using CustomerSupport.Domain.Enums;

namespace CustomerSupport.Domain.Entities;

public class Ticket
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public TicketStatus Status { get; set; } = TicketStatus.Open;
    public TicketPriority Priority { get; set; } = TicketPriority.Medium;
    public int CategoryId { get; set; }
    public int CreatedByUserId { get; set; }
    public int? AssignedToUserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Category? Category { get; set; }
    public User? CreatedByUser { get; set; }
    public User? AssignedToUser { get; set; }

    public ICollection<TicketComment> Comments { get; set; } = [];
    public ICollection<TicketStatusHistory> StatusHistory { get; set; } = [];
}

File: frontend/src/types/ticket.ts
export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: string;
  priority: string;
  categoryId: number;
  categoryName: string;
  createdByUserId: number;
  createdByUserName: string;
  assignedToUserId: number | null;
  assignedToUserName?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface PagedResult<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;

  limit?: number;
  offset?: number;
}

export interface TicketComment {
  id: number;
  ticketId: number;
  userId: number;
  userName: string;
  content: string;
  createdAt: string;
}

export interface TicketStatusHistoryItem {
  id: number;
  ticketId: number;
  oldStatus: string;
  newStatus: string;
  changedByUserId: number;
  changedByUserName: string;
  changedAt: string;
}

export interface Category {
  id: number;
  name: string;
  description: string;
}

export interface CreateTicketRequest {
  title: string;
  description: string;
  categoryId: number;
  createdByUserId: number;
  assignedToUserId: number | null;
  priority: number;
}